server/app/services/rpc/node_service_pod_handler.rb

```diff
@@ -1,21 +1,39 @@
 module Rpc
   class NodeServicePodHandler
+    include Logging
+
+    @@lru_cache ||= LruRedux::ThreadSafeCache.new(1000)
```

Comment:
Doesn't have to be, but there can be multiple instances of `NodeServicePodHandler`, one per grid. So I thought it would keep the cache mem usage at minimum by having it as class variable.